platformio.ini

```diff
@@ -737,42 +732,39 @@ debug_init_break  =
 [env:BIGTREE_GTR_V1_0]
 platform          = ststm32@>=5.7.0
```

Comment:
I'm not entirely sure what we should do about the ststm32 version. Most environments do not specify a version, while others do. As far as being able to build properl with the upgraded dependency, 5.7 is _probably_ the minimum version.

I believe that if we force everything to 6.1, we could remove the `platform_packages` line entirely.

Reply:
I don't think individual boards should be setting ststm32 versions - the HAL is the thing that's linking to a given version.In fact, when PIO finally updates the ST package to include the newest timer functions, the HAL will only compile with the latest (again).

Reply:
@xC0000005 I'm not sure exactly which timer changes you are referring to. The component I am upgrading here is where the timers live, so I might already have them. I mentioned ststm32 6.1 because that would by default request the framework version I am forcing in this PR.

I am aware of changes that went into the HardwareTimers back around November (When we originally worked on and copied in SoftwareSerial).

This upgrade didn't break any compilation in the HAL I am aware of, but there are some changes in those HardwareTimers that will allow us to simplify some things in our HAL, in particular with regards to setting interrupt priorities and pausing/resuming timers.

Are there _other_ changes that may have changed behavior? I don't know. That's why I was hoping to get some people to try it out with whatever their configurations may be. I don't have any of my boards installed right now to actually test anything thoroughly.

Reply:
I just tested this on the bench and there are definitely timer issues.
The watchdog isn't working properly, and resets just as the info screen appears.
Attempting to use Software Serial hangs the board on boot.